pkg/envoy/xds_server.go

```diff
@@ -227,13 +227,13 @@ type xdsServerConfig struct {
 }
 
 // newXDSServer creates a new xDS GRPC server.
-func newXDSServer(logger *slog.Logger, restorerPromise promise.Promise[endpointstate.Restorer], ipCache IPCacheEventSource, localEndpointStore *LocalEndpointStore, config xdsServerConfig, secretManager certificatemanager.SecretManager) *xdsServer {
+func newXDSServer(logger *slog.Logger, initialPoliciesComputedPromise promise.Promise[endpointstate.InitialPoliciesComputed], ipCache IPCacheEventSource, localEndpointStore *LocalEndpointStore, config xdsServerConfig, secretManager certificatemanager.SecretManager) *xdsServer {
```

Comment:
nit: also verbose here - `initialPoliciesComputedPromise` could just be `policiesComputed` or `initialPolicies` IMO, and since you `.Await` them - `initialPolicies.Await(ctx)` reads clear enough I think.